### PR TITLE
Make GESTURE_RESPONSE_DISTANCE_* configurable.

### PR DESCRIPTION
**Pull request by miyabi**
## Motivation

Default `CardStack`'s `GESTURE_RESPONSE_DISTANCE_HORIZONTAL` value is hard coded and too short, so the gesture conflicts with iOS's left edge swipe or other gestures.
My suggestion is to make `GESTURE_RESPONSE_DISTANCE_HORIZONTAL` (and `GESTURE_RESPONSE_DISTANCE_VERTICAL`) configurable via `StackNavigator`'s `StackNavigatorConfig`.

Related issues: #1655 and #1981 

## Test Plan

`gestureResponseDistance` option is added to `StackNavigatorConfig` of `StackNavigator `.
(https://reactnavigation.org/docs/navigators/stack)

If this option is set, the specified value is used as `GESTURE_RESPONSE_DISTANCE_HORIZONTAL` (or `GESTURE_RESPONSE_DISTANCE_VERTICAL`).
If not, the default `GESTURE_RESPONSE_DISTANCE_HORIZONTAL` (or `GESTURE_RESPONSE_DISTANCE_VERTICAL`) is used.

Here is the demo:

### CODE REVIEW

**Comment by aaronleo28** [comment]
When will this functionality be released?

File: src/views/CardStack/CardStack.js [hunk 1]
```diff
@@ -272,9 +273,11 @@ class CardStack extends Component {
         // Measure the distance from the touch to the edge of the screen
         const screenEdgeDistance = currentDragPosition - currentDragDistance;
         // Compare to the gesture distance relavant to card or modal
-        const gestureResponseDistance = isVertical
-          ? GESTURE_RESPONSE_DISTANCE_VERTICAL
-          : GESTURE_RESPONSE_DISTANCE_HORIZONTAL;
+        const gestureResponseDistance =
+          this.props.gestureResponseDistance ||
```
Comment (aaronleo28) [inline src/views/CardStack/CardStack.js]: maybe `gestureResponseDistance` could be an object containing `vertical` and `horizontal` properties so you can individually set response distances for vertical and horizontal?

Comment (aaronleo28) [inline src/views/CardStack/CardStack.js]:
```
const gestureResponseDistance = isVertical
    ? this.props.gestureResponseDistance.vertical || GESTURE_RESPONSE_DISTANCE_VERTICAL
    : this.props.gestureResponseDistance.horizontal || GESTURE_RESPONSE_DISTANCE_HORIZONTAL;
```

**Comment by miyabi** [comment]
@aaronleo28 Thank you for your suggestion. I modified as you suggested.
Here is a new demo:

**Comment by miyabi** [comment]
I noticed `gesturesEnabled` is defined in Screen Navigation Options, so I think that `gestureResponseDistance` should be moved to the same place.

**Comment by miyabi** [comment]
I have moved `gestureResponseDistance` to NavigationStackScreenOptions and added documentation.

**Comment by elliscwc** [comment]
Can this change move to master?

**Comment by vonovak** [comment]
@matthamil this PR seems to be good to be merged

**Comment by vonovak** [comment]
@matthamil @spencercarli this PR appears to be good to be merged

**Comment by RSSfeed** [comment]
Any updates on this?

**Comment by vonovak** [comment]
@davepack @kelset @matthamil this PR appears to be good for merge

**Comment by kelset** [comment]
The branch is outdated, once @miyabi updates it I think it's good to be merged ;)

**Comment by miyabi** [comment]
@matthamil @kelset Thanks!

**Pull request merged**